README.md

```diff
@@ -26,6 +26,34 @@ There are five primary directories in this repository:
 The book is built with [mdbook](https://github.com/rust-lang/mdBook) and deployed via
 [github pages](https://pages.github.com/).
 
+## Building This Book Locally 
+
+Building the book requires [mdBook], ideally the same version that
+rust-lang/rust uses in [this file][rust-mdbook]. To get it:
+
+[mdBook]: https://github.com/rust-lang-nursery/mdBook
+[rust-mdbook]: https://github.com/rust-lang/rust/blob/master/src/tools/rustbook/Cargo.toml
+
+```bash
+$ cargo install mdbook --vers [version-num]
+```
+
+To build the book, type:
+
+```bash
+$ mdbook build
+```
+
+The output will be in the `book` subdirectory. To check it out, open up `book/index.html` in
+a web browser.
+
+
+To run the tests within `rust` code blocks _within the book itself_:
```

Comment:
This might not be needed, as code snippets in the book are not expected to compile?

Reply:
agreed. makes sense to add any of this vs assume people know the context

Reply:
They're not expected to compile on their own. They are tested in the full context, but that is independant of mdbook. Best to omit the `mdbook test`.